### PR TITLE
Left nav persistence on route change 

### LINKED ISSUE #433: Component Library Side Nav Item doesn't collaspe consistently and bumps a user to the top of the nav when clicking on a component
### Description
When trying to engage with the component library navigation element to open and collapse the component library there are times where it doesn't work correctly.

### Steps to reproduce

1. Go to https://developer.newrelic.com/
2. Click on  the  component library nav element. 
3. you will see it opens correctly but now cannot be be collapsed as expected. 
4. If you click on another nav element outside of the component library nav element, it then collapses. 

**THEN**
1. When clicking on a component, we take the user back to the top of the navigation. which is confusing, 

### Expected behavior

When ever engaging with this nav element, is should open and collapse as you'd expect.  Also when clicking on a component in the navigation the user should not be bumped to the top of the navigation.

### PR DESCRIPTION
**Pull request by LizBaker**
## Description
Currently when a user navigates to a new page via the left nav, it refreshes the whole page, including the nav, which results in it "scrolling" back to the top. This PR addresses that behavior and persists the nav position regardless of page changes. 
It also includes a refactoring of the component library subnav to include other subnavs instead of one long list.

## Reviewer Notes
Try clicking around the nav and observe it maintaining it's position even when the page changes 🎉 

## Related Issue(s) / Ticket(s)
Closes #433 

## Screenshot(s)
Here's the new style for the component library subnav
Screen Shot 2020-07-15 at 6 24 18 PM

### CODE REVIEW

**Comment by jpvajda** [comment]
Feedback per our discussion with @djsauble 

1. When a user traverses the navigation and then switches to another top level or sub level nav element collapse the previous nav section they were in.

**Comment by djsauble** [comment]
@jpvajda Close. When the user navigates to a page, we should collapse any nodes that aren't ancestors of the new page.

So, for example, if you've manually expanded multiple sections under the component library, then navigate to (for example) Tooltip, only Overlay, Component library, and Explore docs should be expanded, because they're all ancestors of the Tooltip page.

File: src/components/NavigationItems.js [hunk 1]
```diff
@@ -84,14 +90,37 @@ const NavIcon = ({ page }) => {
 };
 
 const NavItem = ({ page, depthLevel, searchTerm, filteredPageNames }) => {
+  const usePrevious = (value) => {
```
[the remark this answers is not in the document]
Comment (LizBaker) [reply src/components/NavigationItems.js]: I actually ended up getting rid of this (though now that things are being reworked maybe it'll come back 😂 )

[the remark this answers is not in the document]
Comment (LizBaker) [reply src/components/NavigationItems.js]: it's back! and pulled out into the hook folder

**Pull request merged**